src/Lucene.Net/Index/SegmentReader.cs

```diff
@@ -191,10 +191,16 @@ internal Directory GetCFSReader()
 					return cfsReader;
 				}
 			}
-			
+
 			internal TermInfosReader GetTermsReader()
-			{
-				lock (this)
+            {
+                // This is initialized in the constructor.
+                // The only time it isn't initialized is when termsIndexDivisor equals -1, which we don't use.

```

Comment:
What we don't use? Can you elaborate on that?

Reply:
The `TermInfosReader` holds an array of sorted, cached terms and their positions (note that this cache does not include all terms) to efficiently locate the correct term during searches.

The algorithm is roughly the following:
We use binary search to find the position of the closest term within the sorted cache.
https://github.com/ravendb/lucenenet/blob/34bc333c964b47e4c8069e726ba1d316d07eb664/src/Lucene.Net/Index/TermInfosReader.cs#L194
We then iterate until we find the exact match.
https://github.com/ravendb/lucenenet/blob/34bc333c964b47e4c8069e726ba1d316d07eb664/src/Lucene.Net/Index/SegmentTermEnum.cs#L184

The size of the cache is determined by the `termsIndexDivisor`. A larger number creates a smaller cache, requiring more time to find the correct term but reducing memory usage.
The default value is `1`.

Reply:
It's hard for me understand the answer. I was rather expecting you to tell my why is it the case that in our scenarios `termsIndexDivisor` is never `-1`